test/SILGen/opaque_values_silgen.swift

```diff
@@ -8,6 +8,22 @@ protocol P {
   var x : Int { get }
 }
 
+func hasVarArg(_ args: Any...) {}
+
+// Test that we still use addresses when dealing with array initialization
+// ---
+// CHECK-LABEL: sil @_TF20opaque_values_silgen10callVarArgFT_T_ : $@convention(thin) () -> () {
+// CHECK: %[[APY:.*]] = apply %{{.*}}<Any>(%{{.*}}) : $@convention(thin) <τ_0_0> (Builtin.Word) -> (@owned Array<τ_0_0>, Builtin.RawPointer)
+// CHECK: %[[BRW:.*]] = begin_borrow %[[APY]]
+// CHECK: %[[TPL:.*]] = tuple_extract %[[BRW]] : $(Array<Any>, Builtin.RawPointer), 1
+// CHECK: %[[PTR:.*]] = pointer_to_address %[[TPL]] : $Builtin.RawPointer to [strict] $*Any
```

Comment:
Can you add more to the test case for ownership. Specifically, can you make sure that we are ending the borrow and properly destroying APY?